fastlane/Fastfile

```diff
@@ -1240,6 +1367,11 @@ def prompt_user_for_app_store_connect_credentials
   ENV['FASTLANE_USER'] = CredentialsManager::AccountManager.new.user
 end
 
+def switch_to_user_branch(version, message)
+  UI.user_error!("Release branch for version #{version} doesn't exist. Abort.") unless Fastlane::Helper::GitHelper.checkout_and_pull(release: version)
+  message << "Looking at branch release/#{version} as requested by user. Detected version: #{current_build_code}.\n"
+end
```

Comment:
Oh ok my bad, it does call `checkout_and_pull`, it's just that this is kinda hidden within the `unless` clause (I didn't expect any "action with side effect" to be in the `unless` clause, as opposed to just a boolean value, that's why I missed it)

Maybe to help readability we should rewrite this like that?
```suggestion
def switch_to_user_branch(version, message)
  success = Fastlane::Helper::GitHelper.checkout_and_pull(release: version)
  UI.user_error!("Release branch for version #{version} doesn't exist. Abort.") unless success
  message << "Looking at branch release/#{version} as requested by user. Detected version: #{current_build_code}.\n"
end
```

Also, I'm not a fan of having a method that modifies one of its input parameters (`message`), as opposed to return it as a new value. Doesn't feel very idiomatic of Ruby.

Reply:
Addressed in